src/ReplicatedStorage/Client/Tool/Gun/init.lua

```diff
@@ -275,11 +488,24 @@ function Gun:OnActivated()
 	end)
 end
 
+function Tool:OnEquipped()
+	if self._trove then
+		self._trove:Destroy()
+	end
+	self._trove = Trove.new()
+	self:_createGui()
+	self.Active = false
+	self:ShowReloadBar()
+end
+
 function Tool:OnDeactivated()
 	self.Active = false
 end
 
 function Tool:OnUnequipped()
+	if self._trove then
+		self._trove:Destroy()
+	end
```

Comment:
trove:Clean() is for manual use - :Destroy() only exists so Trove implements the Destroyable duck-type.